### PR TITLE
✨ Add first definition of logical.Name

### PR DESCRIPTION
**Pull request by vincepri**

### CODE REVIEW

File: object.go [hunk 1]
```diff
@@ -0,0 +1,12 @@
+package logical
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// Annotation defines the logical annotation that can be attached to a Kubernetes object
+// to determine the logical cluster name the object belongs to.
+const Annotation = "logical.sigs.x-k8s.io/cluster"
+
+// NameFromObject returns the logical cluster name from the given Kubernetes object, if any.
+func NameFromObject(obj metav1.Object) Name {
```
Comment (ncdc) [inline object.go]: Would not use metav1.Object to minimize dependencies. Suggest instead:

```go
type AnnotatedObject interface {
  GetAnnotations() map[string]string
}
```

Feel free to use some other name besides AnnotatedObject 😄

Comment (vincepri) [reply object.go]: Nice and lean now

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/env bash
```
Comment (ncdc) [inline Makefile]: Can probably remove the shebang?

Comment (vincepri) [reply Makefile]: done

**Review by ncdc — APPROVED** [review]

**Pull request merged**